templates/workspace_services/azureml/template_schema.json

```diff
@@ -1,11 +1,146 @@
+                  "description": "Azure Active Directory",
+                  "source_addresses": [
+                    "{{ resource.properties.workspace_services_subnet_address_prefix }}"
+                  ],
+                  "destination_addresses": [
+                    "AzureActiveDirectory"
+                  ],
+                  "destination_ports": [
+                    "443",
+                    "80"
+                  ],
+                  "protocols": [
+                    "TCP"
+                  ]
+                },
+                {
+                  "name": "AzureML_Dependancies",
+                  "description": "AzureML Dependancies",
+                  "source_addresses": [
+                    "{{ resource.properties.workspace_services_subnet_address_prefix }}"
+                  ],
+                  "destination_addresses": [
+                    "AzureActiveDirectory",
+                    "AzureResourceManager",
+                    "MicrosoftContainerRegistry"
+                  ],
+                  "destination_ports": [
+                    "443"
+                  ],
+                  "protocols": [
+                    "TCP"
+                  ]
+                },
+                {
+                  "name": "AzureML_Storage",
+                  "description": "AzureML Storage",
+                  "source_addresses": [
+                    "{{ resource.properties.workspace_services_subnet_address_prefix }}"
+                  ],
+                  "destination_addresses": [
+                    "{{ resource.properties.storage_tag }}"
+                  ],
+                  "destination_ports": [
+                    "443",
+                    "445"
+                  ],
+                  "protocols": [
+                    "TCP"
+                  ]
+                }
+              ]
+            }
+          },
+          {
+            "name": "rule_collections",
+            "type": "array",
+            "arraySubstitutionAction": "replace",
+            "arrayMatchField": "name",
+            "value": {
+              "name": "arc_svc_{{ resource.id }}_azureml",
+              "action": "Allow",
+              "rules": [
+                {
+                  "name": "AzureML",
+                  "description": "AzureML rules",
+                  "source_addresses": [
+                    "{{ resource.properties.workspace_services_subnet_address_prefix }}"
+                  ],
+                  "target_fqdns": [
+                    "aadcdn.msftauth.net",
+                    "ml.azure.com"
+                  ],
+                  "protocols": [
+                    {
+                      "port": "443",
+                      "type": "Https"
+                    }
+                  ]
+                }
+              ]
+            }
+          }
+        ]
+      }
     ],
-    "properties": {
-    }
+    "upgrade": [
+      {
+        "stepId": "main"
+      }
+    ],
+    "uninstall": [
```

Comment:
as we discussed - let's put in the uninstall logic to remove the FW rule on resource uninstall (and possibly upgrade if anything could change)